x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elser/ElserInternalService.java

```diff
@@ -355,25 +356,22 @@ private static ElserMlNodeTaskSettings taskSettingsFromMap(TaskType taskType, Ma
         return ElserMlNodeTaskSettings.DEFAULT;
     }
 
-    private ChunkedSparseEmbeddingResults translateChunkedResults(List<InferenceResults> inferenceResults) {
-        if (inferenceResults.size() != 1) {
-            throw new ElasticsearchStatusException(
-                "Expected exactly one chunked sparse embedding result",
-                RestStatus.INTERNAL_SERVER_ERROR
-            );
-        }
+    private List<ChunkedInferenceServiceResults> translateChunkedResults(List<InferenceResults> inferenceResults) {
+        var translated = new ArrayList<ChunkedInferenceServiceResults>();
 
-        if (inferenceResults.get(0) instanceof ChunkedTextExpansionResults mlChunkedResult) {
-            return ChunkedSparseEmbeddingResults.ofMlResult(mlChunkedResult);
-        } else {
-            throw new ElasticsearchStatusException(
-                "Expected a chunked inference [{}] received [{}]",
-                RestStatus.INTERNAL_SERVER_ERROR,
-                ChunkedTextExpansionResults.NAME,
-                inferenceResults.get(0).getWriteableName()
-            );
+        for (var inferenceResult : inferenceResults) {
+            if (inferenceResult instanceof ChunkedTextExpansionResults mlChunkedResult) {
+                translated.add(ChunkedSparseEmbeddingResults.ofMlResult(mlChunkedResult));
+            } else {
+                throw new ElasticsearchStatusException(
+                    "Expected a chunked inference [{}] received [{}]",
+                    RestStatus.INTERNAL_SERVER_ERROR,
+                    ChunkedTextExpansionResults.NAME,
+                    inferenceResults.get(0).getWriteableName()
```

Comment:
Should this be `inferenceResult.getWriteableName()`?

Reply:
👍 good catch